ConsoleAppForTesting/ThinkWhenEat.cs

```diff
@@ -0,0 +1,94 @@
+﻿namespace ConsoleAppForTesting
+{
+    using System.Threading;
+
+    /// <summary>Task about dining philosophers problem</summary>
+    public class PhilosophersCafe
+    {
+        /// <summary>Gets the philosophers.</summary>
+        /// <value>The philosophers.</value>
+        private Thread[] Philosophers { get; }
+        /// <summary>The forks for philosophers.</summary>
+        private SemaphoreSlim[] forks;
+        /// <summary>The time during which the philosopher eats.</summary>
+        private int timeForPhilosopherLunch;
+        /// <summary>The time during which the philosopher thinks.</summary>
+        private int timeForPhilosopherThink;
+        /// <summary>The count of dinner intake for each of philosophers.</summary>
+        private int countOfLunchIntake;
+        /// <summary>Gets the states of each of philosophers.</summary>
+        /// <value>The states of each philosophers.</value>
+        public string[] StatesOfPhilosophers { get;}
+        /// <summary>The count of philosophers.</summary>
+        private int countOfPhilosophers;
+
+        /// <summary>Initializes a new instance of the <see cref="PhilosophersCafe"/> class.</summary>
+        /// <param name="countOfPhilosophers">The count of philosophers.</param>
+        /// <param name="countOfLunchIntake">The count of dinner intake.</param>
+        /// <param name="timeForLunch">The time during which the philosopher eats.</param>
+        /// <param name="timeForThink">The time during which the philosopher thinks.</param>
+        public PhilosophersCafe(int countOfPhilosophers, int countOfLunchIntake, int timeForLunch, int timeForThink)
+        {
+            this.countOfPhilosophers = countOfPhilosophers;
+            Philosophers = new Thread[countOfPhilosophers];
+            forks = new SemaphoreSlim[countOfPhilosophers];
+            StatesOfPhilosophers = new string[countOfPhilosophers];
+            timeForPhilosopherLunch = timeForLunch;
+            timeForPhilosopherThink = timeForThink;
+            this.countOfLunchIntake = countOfLunchIntake;
+
+            for (var indexOfPhilosopher = 0; indexOfPhilosopher < countOfPhilosophers; indexOfPhilosopher++)
+            {
+                Philosophers[indexOfPhilosopher] = new Thread(TimeToDinner);
+                forks[indexOfPhilosopher] = new SemaphoreSlim(1,1);
+                StatesOfPhilosophers[indexOfPhilosopher] += $"Philosopher# {indexOfPhilosopher} is coming into cafe || ";
+                Philosophers[indexOfPhilosopher].IsBackground = true;
+                Philosophers[indexOfPhilosopher].Start(indexOfPhilosopher);
```

Comment:
It's not good idea to perform actions other than initialization in constructor, so please move start to somewhere else.